app/src/main/java/com/example/calculator/mvp/presenter/CalculatorPresenter.java

```diff
@@ -0,0 +1,22 @@
+package com.example.calculator.mvp.presenter;
+
+import com.example.calculator.mvp.contract.CalculatorContract;
+import com.example.calculator.mvp.model.CalculatorModel;
+import com.example.calculator.mvp.view.CalculatorView;
+
+public class CalculatorPresenter implements CalculatorContract.Presenter {
+
+    private CalculatorModel model;
```

Comment:
these variables must be of the contract type

Reply:
Yes! solved.